--- mpd-0.17.2/src/input/cdio_paranoia_input_plugin.c~	2012-09-30 12:22:57.000000000 +0200
+++ mpd-0.17.2/src/input/cdio_paranoia_input_plugin.c	2013-02-03 22:28:42.169606736 +0100
@@ -35,7 +35,12 @@
 #include <glib.h>
 #include <assert.h>
 
+#include <cdio/cdio.h> 
+#if LIBCDIO_VERSION_NUM <= 83
 #include <cdio/paranoia.h>
+#else
+#include <cdio/paranoia/paranoia.h>
+#endif
 #include <cdio/cd_types.h>
 
 struct input_cdio_paranoia {
