diff -urN mpd-0.20.3.orig/configure.ac mpd-0.20.3/configure.ac
--- mpd-0.20.3.orig/configure.ac	2017-01-30 22:46:56.376450224 +0100
+++ mpd-0.20.3/configure.ac	2017-01-30 22:47:26.799710230 +0100
@@ -922,7 +922,7 @@
 
 dnl --------------------------------- musepack --------------------------------
 
-MPD_ENABLE_AUTO_LIB(mpc, MPCDEC, mpcdec, mpc_demux_init, [-lmpcdec], [],
+MPD_ENABLE_AUTO_LIB(mpc, MPCDEC, mpcdecsv8, mpc_demux_init, [-lmpcdecsv8], [],
 	[Musepack decoder plugin], [libmpcdec not found])
 
 dnl -------------------------------- Ogg Tremor -------------------------------
