--- mpd-0.24.9/src/Main.cxx.orig	2026-03-11 11:54:28.000000000 +0100
+++ mpd-0.24.9/src/Main.cxx	2026-03-12 01:27:56.922373764 +0100
@@ -695,15 +695,6 @@
 int
 main(int argc, char *argv[]) noexcept
 try {
-#ifdef __linux__
-	if (ProcStatusThreads() > 1)
-		/* threads created by libraries before main() can
-		   cause all sorts of bugs that are difficult to
-		   analyze; bail out quickly and refuse to run if we
-		   detect such a thing */
-		throw "Background threads were detected before MPD could initialize. This is likely caused by a misbehaving library. Aborting to prevent erroneous behavior. Please report.";
-#endif
-
 	AtScopeExit() { log_deinit(); };
 
 #ifdef _WIN32
